blocks/bricks/sequence_generators.py

```diff
@@ -257,6 +266,13 @@ def cost_matrix(self, application_call, outputs, mask=None, **kwargs):
             mask=mask, return_initial_states=True, as_dict=True,
             **dict_union(inputs, states, contexts))
 
+        # Run the language model
+        if self.language_model:
+            lm_arguments = self.language_model.apply(
+                outputs, mask=mask, as_dict=True)
```

Comment:
Imagine, that we have so powerful LM that it can always reconstruct the output. In this case the cost is always 0 and the AM won't receive any gradient.